--- a/setup/src/main/java/dev/jdtech/jellyfin/setup/presentation/addresses/ServerAddressesState.kt
+++ b/setup/src/main/java/dev/jdtech/jellyfin/setup/presentation/addresses/ServerAddressesState.kt
@@ -2,6 +2,4 @@ package dev.jdtech.jellyfin.setup.presentation.addresses
 
 import dev.jdtech.jellyfin.models.ServerAddress
 
-data class ServerAddressesState(
-    val addresses: List<ServerAddress> = emptyList(),
-)
+data class ServerAddressesState(val addresses: List<ServerAddress> = emptyList())
