--- a/app/src/main/java/fr/neamar/kiss/utils/ClipboardUtils.java
+++ b/app/src/main/java/fr/neamar/kiss/utils/ClipboardUtils.java
@@ -4,16 +4,18 @@ import android.content.ClipData;
 import android.content.ClipboardManager;
 import android.content.Context;
 
+import androidx.annotation.NonNull;
+import androidx.core.content.ContextCompat;
+
 public class ClipboardUtils {
-	public static void setClipboard(Context context, String text) {
-		ClipboardManager clipboard;
-		if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
-			clipboard = context.getSystemService(ClipboardManager.class);
-		} else {
-			clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
-		}
 
-		ClipData clip = ClipData.newPlainText("Copied Text", text);
+	public static void setClipboard(@NonNull Context context, String text) {
+		setClipboard(context, "Copied Text", text);
+	}
+
+	public static void setClipboard(@NonNull Context context, String label, String text) {
+        ClipboardManager clipboard = ContextCompat.getSystemService(context, ClipboardManager.class);
+		ClipData clip = ClipData.newPlainText(label, text);
 		clipboard.setPrimaryClip(clip);
 	}
 }
