--- a/app/src/main/java/com/junkfood/seal/util/SponsorData.kt
+++ b/app/src/main/java/com/junkfood/seal/util/SponsorData.kt
@@ -8,23 +8,27 @@ data class SponsorData(
 )
 
 @Serializable
-data class User(
-    val sponsorshipsAsMaintainer: SponsorshipsAsMaintainer
+data class Data(
+    val viewer: Viewer
 )
 
 @Serializable
-data class SponsorshipsAsMaintainer(
-    val nodes: List<Node>
+data class Viewer(
+    val sponsorshipsAsMaintainer: SponsorshipsAsMaintainer
 )
 
 @Serializable
-data class Data(
-    val user: User
+data class SponsorshipsAsMaintainer(
+    val nodes: List<SponsorShip>
 )
 
+
 @Serializable
 data class SponsorEntity(
-    val login: String, val name: String?
+    val login: String,
+    val name: String? = null,
+    val websiteUrl: String? = null,
+    val socialAccounts: SocialAccounts? = null
 )
 
 @Serializable
@@ -33,6 +37,14 @@ data class Tier(
 )
 
 @Serializable
-data class Node(
-    val sponsorEntity: SponsorEntity, val tier: Tier
-)
\ No newline at end of file
+data class SponsorShip(
+    val sponsorEntity: SponsorEntity, val tier: Tier? = null
+)
+
+@Serializable
+data class SocialAccounts(
+    val nodes: List<SocialAccount>? = null
+)
+
+@Serializable
+data class SocialAccount(val displayName: String?, val url: String?)
\ No newline at end of file
